packages/http-server/src/server.ts

```diff
@@ -10,63 +11,75 @@ import { IPrismDiagnostic } from '@stoplight/prism-core';
 import { pipe } from 'fp-ts/lib/pipeable';
 import * as TE from 'fp-ts/lib/TaskEither';
 import * as E from 'fp-ts/lib/Either';
+import { MicriHandler } from 'micri';
+
+function searchParamsToNameValues(searchParams: URLSearchParams): IHttpNameValues {
+  const params = {};
+  for (const key of searchParams.keys()) {
+    const values = searchParams.getAll(key);
+    params[key] = values.length === 1 ? values[0] : values;
+  }
+  return params;
+}
+
+function addressInfoToString(address: AddressInfo | string | null) {
+  if (!address) return '';
+  const a = address as AddressInfo;
+  return `http://${a.address}:${a.port}`;
+}
+
+function parseRequestBody(request: IncomingMessage) {
+  // if no body provided then return null instead of empty string
+  if (
+    request.headers['content-type'] === undefined
+    && request.headers['transfer-encoding'] === undefined
+    && (request.headers['content-length'] === '0' || request.headers['content-length'] === undefined)
+  ) {
+    return Promise.resolve(null);
+  }
+
+  if (typeIs(request, ['application/json', 'application/*+json'])) {
+    return json(request);
+  } else {
+    return text(request);
+  }
+}
 
 export const createServer = (operations: IHttpOperation[], opts: IPrismHttpServerOpts): IPrismHttpServer => {
   const { components, config } = opts;
 
-  const server = fastify({
-    logger: components.logger,
-    disableRequestLogging: true,
-    modifyCoreObjects: false,
-  });
-
-  if (opts.cors) server.register(fastifyCors, { origin: true, credentials: true });
-
-  server.addContentTypeParser('*', { parseAs: 'string' }, (req, body, done) => {
-    if (typeIs(req, ['application/*+json'])) {
-      try {
-        return done(null, JSON.parse(body));
-      } catch (e) {
-        return done(e);
-      }
-    }
+  const handler: MicriHandler = async (request, reply) => {
+    const {
+      url,
+      method,
+      headers,
+      socket,
+    } = request;
 
-    if (typeIs(req, ['application/x-www-form-urlencoded'])) {
-      return done(null, body);
+    if (!url) {
+      return TE.left(new Error('Missing URL in request!'));
```

Comment:
Ok this condition is never going to happen.

Follow up question: "Why is the URL then nullable?"

Because the same object `HttpServer.IncomingMessage` is used also for a ServerRequest which has not such property.

In our case — `HttpServer.IncomingMessage` is coming from a `ClientRequest` — which always has the URL property.

Source: https://nodejs.org/dist/latest-v12.x/docs/api/http.html#http_class_http_incomingmessage

Reply:
Said some guy about 640kb of memory ;)

Anyway, I consider making req.url! with a proper comment, WDYT?

Reply:
Fair!